tony-history-server/app/utils/ParserUtils.java

```diff
@@ -0,0 +1,88 @@
+package utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.gson.JsonSyntaxException;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import models.JobConfig;
+import models.JobMetadata;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import play.Logger;
+import play.libs.Json;
+
+import static utils.HdfsUtils.*;
+
+/**
+ * The class handles parsing different file format
+ */
+public class ParserUtils {
```

Comment:
Create a `private ParserUtils() { }` constructor in utility classes to prevent instantiation. See _Effective Java_ Item 3.

Reply:
I'm assuming this would also apply for other utils class?

Reply:
Yes, that's right.

Reply:
Fixed: https://github.com/linkedin/TonY/pull/62/commits/7a29abf098575b18b7b82256b6603d8d28f475f8